src/components/visualizations/chart/colorFactory.ts

```diff
@@ -99,14 +108,14 @@ export abstract class ColorStrategy implements IColorStrategy {
         });
     }
 
-    protected abstract createColorMapping(
+    protected abstract createColorAssignment(
         colorPalette: IColorPalette,
         colorMapping: IColorMapping[],
         measureGroup: MeasureGroupType,
         viewByAttribute: any,
         stackByAttribute: any,
         afm: AFM.IAfm
-    ): IColorAssignment[];
+    ): { fullColorAssignment: IColorAssignment[], outputColorAssignment?: IColorAssignment[] };
```

Comment:
Define interface for return type { fullColorAssignment: IColorAssignment[], outputColorAssignment?: IColorAssignment[] } something like  IColorAssignmentType or wrapper or ...